wire-ios-sync-engine/Tests/Source/Calling/CallKitManagerTests.swift

```diff
@@ -945,4 +962,75 @@ class CallKitManagerTest: DatabaseTest {
         XCTAssertEqual(self.callKitProvider.lastEndedReason, .remoteEnded)
     }
 
+    // MARK: - Call Rejection
+
+    func test_itSyncsBeforeRejectingCall() {
+        // given
+        let conversation = conversation()
+        let otherUser = otherUser(moc: uiMOC)
+
+        sut.reportIncomingCall(from: otherUser, in: conversation, hasVideo: false)
+
+        guard let callKitCall = sut.callRegister.lookupCall(by: conversation) else {
+            XCTFail("no call handle")
+            return
+        }
+
+        let mockEndCallAction = MockEndCallAction(call: callKitCall.id)
+
+        // when
+        sut.provider(callKitProvider, perform: mockEndCallAction)
+
+        // then
+        XCTAssertEqual(mockCallKitManagerDelegate.lookupConversationAndSyncCalls, 1)
+    }
+
+    func test_itUnregistersCallAfterRejecting() {
+        // given
+        let conversation = conversation()
+        let otherUser = otherUser(moc: uiMOC)
+
+        sut.reportIncomingCall(from: otherUser, in: conversation, hasVideo: false)
+
+        guard let callKitCall = sut.callRegister.lookupCall(by: conversation) else {
+            XCTFail("no call handle")
+            return
+        }
+
+        mockCallKitManagerDelegate.mockConversations = [
+            callKitCall.handle: conversation
+        ]
+
+        let mockEndCallAction = MockEndCallAction(call: callKitCall.id)
+
+        // when
+        sut.provider(callKitProvider, perform: mockEndCallAction)
+
+        // then
+        XCTAssertFalse(sut.callRegister.callExists(for: callKitCall.id))
+    }
+
+    func test_itUnregistersCallAfterRejecting_ConversationLookupFailed() {
+        // given
+        let conversation = conversation()
+        let otherUser = otherUser(moc: uiMOC)
+
+        sut.reportIncomingCall(from: otherUser, in: conversation, hasVideo: false)
+
+        guard let callKitCall = sut.callRegister.lookupCall(by: conversation) else {
+            XCTFail("no call handle")
+            return
+        }
+
+        mockCallKitManagerDelegate.mockConversations = [:]
```

Comment:
Have no mock conversation is what causes it to fail?

Reply:
Yes